lib/btree.js

```diff
@@ -0,0 +1,247 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+// And [degree - 1 ... 2 * degree - 1] + 1 "empty" elements
+
+class Element {
+  constructor(key, data, child = null) {
+    this.key = key;
+    this.data = data;
+    this.child = child;
+  }
+}
+
+const empty = child => new Element(undefined, undefined, child);
+
+const splitNode = (parent, index) => {
+  const node = parent[index].child;
+  const len = node.length;
+  const newLeftNode = node.splice(0, len / 2 - 1); // First half
+  const mid = node.shift();
+  newLeftNode.push(empty(mid.child));
+  mid.child = newLeftNode;
+  parent.splice(index, 0, mid);
+  return parent;
+};
+
+const isLeaf = node => !node[0].child;
+
+const binarySearch = (node, key) => {
+  let start = 0;
+  let end = node.length - 1;
+  while (start <= end) {
+    const i = (start + end) >> 1;
+    const item = node[i];
+    const itemValue = item.key;
+    if (key > itemValue) start = i + 1;
+    // the biggest element in the node has key === undefined, therefore,
```

Comment:
How about just adding the check if `key === undefined` instead of using implicit type conversion in strange looking value checks like the one below?